### PR TITLE
Rf/zenko 250 portmdsearchtest

### PR DESCRIPTION
**Pull request by alexanderchan-scality**
Documentation: #1220

### CODE REVIEW

File: tests/functional/aws-node-sdk/test/mdSearch/basicSearch.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
+const s3Client = require('./utils/s3SDK');
+const { runAndCheckSearch, runIfMongo } = require('./utils/helpers');
+
+const objectKey = 'findMe';
+const hiddenKey = 'leaveMeAlone';
+const userMetadata = { food: 'pizza' };
+const updatedUserMetadata = { food: 'cake' };
+
+runIfMongo('Basic search', () => {
```
Comment (philipyoo) [inline tests/functional/aws-node-sdk/test/mdSearch/basicSearch.js]: this is cool 👍

File: tests/functional/aws-node-sdk/test/mdSearch/versionEnabledSearch.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            return s3Client.putBucketVersioning({ Bucket: bucketName,
+                VersioningConfiguration }, err => {
+                if (err) {
+                    return done(err);
+                }
+                return s3Client.putObject({ Bucket: bucketName,
+                    Key: masterKey, Metadata: userMetadata },
+                    err => {
+                        // give ingestion pipeline some time
+                        // setTimeout(() => done(err), 45000);
```
Comment (philipyoo) [inline tests/functional/aws-node-sdk/test/mdSearch/versionEnabledSearch.js]: just curious, why were the timeouts originally used? Is this the same as the timeout applied in the npm run command `ft_search`?

File: lib/api/apiUtils/bucket/validateSearch.js [hunk 3]
```diff
@@ -40,7 +40,14 @@ function _validateTree(whereClause, possibleAttributes) {
 function validateSearchParams(searchParams) {
     let ast;
     try {
-        ast = parser.parse(searchParams);
+        ast = parser.parse(searchParams, (operator, operands) => {
+            switch (operator) {
```
Comment (philipyoo) [inline lib/api/apiUtils/bucket/validateSearch.js]: why not just an if/else?

File: tests/functional/aws-node-sdk/test/mdSearch/versionEnabledSearch.js (hunk 2 above)
Comment (alexanderchan-scality) [reply tests/functional/aws-node-sdk/test/mdSearch/versionEnabledSearch.js]: The current implementation of MD search queries the mongo MD backend, and, as such, it does not require the wait time. (@rahulreddy confirmation/elaborate this statement?)

`setTimeout` was originally used here when MD search was using Clueso. As for how it relates to the npm command timeout: the npm timeout is set to a larger number to make sure this timeout wait does not expire the test.

**Comment by vrancurel** [comment]
and the design !

**Comment by alexanderchan-scality** [comment]
I have split the documentation into another PR https://github.com/scality/S3/pull/1220

**Pull request merged**